lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/evaluator/EvaluatorShim.java

```diff
@@ -109,13 +109,24 @@ public void onNext(final RemoteMessage<EvaluatorShimProtocol.EvaluatorShimContro
     switch (command) {
     case LAUNCH_EVALUATOR:
       LOG.log(Level.INFO, "Received a command to launch the Evaluator.");
-      this.onEvaluatorLaunch(remoteMessage.getMessage().getEvaluatorLaunchCommand(),
-          remoteMessage.getMessage().getEvaluatorConfigString());
+      this.threadPool.submit(new Runnable() {
+        @Override
+        public void run() {
+          EvaluatorShim.this.onEvaluatorLaunch(remoteMessage.getMessage().getEvaluatorLaunchCommand(),
+              remoteMessage.getMessage().getEvaluatorConfigString());
+        }
+      });
       break;
 
     case TERMINATE :
-      LOG.log(Level.INFO, "Received a command to terminate.");
-      this.onStop();
+      LOG.log(Level.INFO, "Received a command to terminate the EvaluatorShim.");
+      this.threadPool.submit(new Runnable() {
+        @Override
+        public void run() {
+          EvaluatorShim.this.onStop();
+          EvaluatorShim.this.threadPool.shutdown();
```

Comment:
Minor: put this.theadPool.shutdown into onStop()?

Reply:
Sure.